service/db.hpp

```diff
@@ -80,370 +89,309 @@ namespace raven
-                }
-                DLOG_F(INFO, "json after update: %s", json_data.dump().c_str());
-                this->execute_statement(update_config_text_from_id_statement,
-                    json_data.dump().c_str(), db_id.value());
-            };
-            throw_misuse_if_count_return_zero_for_this_statement(select_count_config_from_id_statement,
-                                                                 db_id.value());
-            execute_statement(select_config_from_id_statement, db_id.value()) >> functor_receive_data;
+            int nb_count = 0;
+            execute_statement(select_count_config_element_statement) >> nb_count;
+            if (!nb_count)
+                throw sqlite::errors::empty(0, select_count_config_element_statement.value());
+            throw_misuse_if_count_return_zero_for_this_statement(select_count_key_statement, config_key.value(), config_key.value());
+            execute_statement(select_config_from_key_statement, config_key.value(), config_key.value()) >> functor_receive_data;
+            state = db_state::good;
+        }
+        catch (const sqlite::errors::empty &error) {
+            DLOG_F(ERROR, "misuse of api or wrong key: %s, from sql: %s", error.what(), error.get_sql().c_str());
+            state = db_state::sql_error;
         }
         catch (const sqlite::errors::misuse &error) {
-            DLOG_F(ERROR, "misuse of api or wrong id: %s, from sql: %s -> [with id = %lu]", error.what(),
-                error.get_sql().c_str(),
-                db_id.value());
-            return request_state::unknown_id;
+            DLOG_F(ERROR, "misuse of api or wrong key: %s, from sql: %s", error.what(), error.get_sql().c_str());
+            state = db_state::unknow_config_key;
         }
         catch (const sqlite::sqlite_exception &error) {
             DLOG_F(ERROR, "error: %s, from sql: %s", error.what(), error.get_sql().c_str());
-            return request_state::db_error;
+            state = db_state::sql_error;
         }
         catch (const std::exception &error) {
             DLOG_F(ERROR, "%s", error.what());
-            return request_state::db_error;
+            state = db_state::fatal_error;
         }
-        return request_state::success;
+        return config_id;
     }
 
-    setting_get_answer setting_get(const setting_get& data, raven::config_id_st db_id) noexcept
+    // TODO get permission from id
+
+    json::json get_config(config_id_st id) noexcept
     {
-        setting_get_answer answer;
+        /*
+         * Get the corresponding config in json format to the id passed as parameter
+         *
+         * In case an error occur, the state will be set accordingly (`unknow_config_id`, `sql_error`, `fatal_error`)
+         *
+         */
+
+        LOG_SCOPE_F(INFO, __PRETTY_FUNCTION__);
+        json::json data;
+        auto functor_receive_data = [&data, id, this](const std::string json_text) { data = json::json::parse(json_text); };
```

Comment:
Is `this` really used ?

Reply:
Nope neither is id, sorry didn't pay attention